### ISSUE TITLE
Single command with arguments

### ISSUE THREAD

**Issue by Shatur** [stevearc/overseer.nvim#20]
Currently the plugin have a lot of standalone commands. But it would be great to provide a single command with subcommands instead.
Like `Overseer open` or `Overseer run`. It just nicer to use with completion.

**Comment by stevearc** [comment]
I agree that having a single `:Overseer` command simplifies the interface, but I think it hides necessary complexity. Some examples why I prefer the multiple-command approach:

- Easier to search/browse available commands (e.g. `:Telescope commands`)
- Integration points are simpler. I just hook directly into `nvim_create_command` instead of needing to create a framework for adding custom subcommands with their own completion, arguments, etc.
- It's intuitive for a user to do `:help OverseerRun` for help with a command. It's not obvious what you would `:help` for to get information about `:Overseer run`.

On top of that, the completion experience seems mostly the same whether you do `:Overseer <Tab>` or `:Overseer<Tab>`

**Comment by Shatur** [comment]
> Easier to search/browse available commands (e.g. :Telescope commands)

That's right.

> Integration points are simpler. I just hook directly into nvim_create_command instead of needing to create a framework for adding custom subcommands with their own completion, arguments, etc.

In my plugins I do it like this: all high level lua functions located in [a single file](https://github.com/Shatur/neovim-session-manager/blob/master/lua/session_manager/init.lua) and then I just [import it](https://github.com/Shatur/neovim-session-manager/blob/6604857365b13bfbcaa7ef377d4e60d2acb0be02/lua/session_manager/subcommands.lua#L1) and [iterate over its functions](https://github.com/Shatur/neovim-session-manager/blob/6604857365b13bfbcaa7ef377d4e60d2acb0be02/lua/session_manager/subcommands.lua#L15) to provide completion. And [here](https://github.com/Shatur/neovim-session-manager/blob/6604857365b13bfbcaa7ef377d4e60d2acb0be02/lua/session_manager/subcommands.lua#L25) I execute the selected command. [Here](https://github.com/Shatur/neovim-session-manager/blob/6604857365b13bfbcaa7ef377d4e60d2acb0be02/plugin/session_manager.lua#L9) is how I use these two functions, a single line. It's super easy after Neovim 0.7 :smile:

> It's intuitive for a user to do :help OverseerRun for help with a command. It's not obvious what you would :help for to get information about :Overseer run.

In Telescope its done via `command.subcommand`. So `:h overseer.run` should work. Users will see it with completion.

> On top of that, the completion experience seems mostly the same whether you do :Overseer <Tab> or :Overseer<Tab>

Not exactly. In my case typing `Over<Tab>` results in `OverseerRun` :( I use nvim-cmp for cmd completion.
But with subcommands it will become just `Overseer` and I can continue typing a subcommand.
